src/main/scala/org/constellation/primitives/RandomTransactionManager.scala

```diff
@@ -173,14 +175,18 @@ class RandomTransactionManager[T](nodeActor: ActorRef, periodSeconds: Int = 10)(
                   .put(TransactionCacheData(tx, valid = true, inMemPool = true))
                   .unsafeRunSync()
 
-                dao
-                  .peerInfo(NodeType.Full)
-                  .unsafeRunSync()
-                  .values
-                  .foreach { peerData ⇒
-                    dao.metrics.incrementMetric("transactionPut")
-                    peerData.client.put("transaction", tx)
-                  }
+                implicit val random = scala.util.Random
+                val contextShift: ContextShift[IO] = IO.contextShift(dao.edgeExecutionContext)
+
+                val rebroadcast = for {
+                  tcd <- dao.transactionGossiping.observe(TransactionCacheData(tx))
+                  peers <- dao.transactionGossiping.selectPeers(tcd)
+                  peerData <- dao.peerInfo(NodeType.Full).map(_.filterKeys(peers.contains).values.toList)
+                  _ <- contextShift.shift *> peerData.traverse(_.client.putAsync("transaction", tcd.transaction))
+                  _ <- dao.metrics.incrementMetricAsync[IO]("transactionGossipingSent")
+                } yield ()
+
+                rebroadcast.unsafeRunSync
```

Comment:
If you don't need result, maybe `runAsyncAndForget` would fit better ?

Reply:
I don't want to do it asynchronous